django/db/migrations/state.py

```diff
@@ -115,6 +115,11 @@ def relations(self):
             self.resolve_fields_and_relations()
         return self._relations
 
+    def update_op_type(self, app_label, model_name, options):
+        model_state = self.models[app_label, model_name]
+        model_state.options = {**model_state.options, **options}
+        self.reload_model(app_label, model_name, delay=True)
```

Comment:
Model should not be aware (and maintain) migration-specific options. I'd keep it in the `ModelState`.